src/AllocationBoundsInference.cpp

```diff
@@ -86,6 +86,17 @@ class AllocationInference : public IRMutator {
                 max = b[i].max;
                 extent = simplify((max - min) + 1);
             }
+            if (bound.modulus.defined()) {
+                internal_assert(bound.remainder.defined());
+                min -= bound.remainder;
+                min = (min / bound.modulus) * bound.modulus;
+                min += bound.remainder;
+                Expr max_plus_one = max + 1;
+                max_plus_one -= bound.remainder;
+                max_plus_one = ((max_plus_one + bound.modulus - 1) / bound.modulus) * bound.modulus;
+                max_plus_one += bound.remainder;
+                extent = simplify(max_plus_one - min);
```

Comment:
Maybe if it were a round-up that would work, but it doesn't work to round
the extent downwards, even after shifting the min but not the max. I'll try
the rounding-up variant.

On Tue, Jul 19, 2016 at 3:32 PM, Dillon Sharlet <notifications@github.com>
wrote:

> In src/AllocationBoundsInference.cpp
> <https://github.com/halide/Halide/pull/1392#discussion_r71436589>:
>
> > @@ -86,6 +86,17 @@ class AllocationInference : public IRMutator {
> >                  max = b[i].max;
> >                  extent = simplify((max - min) + 1);
> >              }
> > +            if (bound.modulus.defined()) {
> > +                internal_assert(bound.remainder.defined());
> > +                min -= bound.remainder;
> > +                min = (min / bound.modulus) * bound.modulus;
> > +                min += bound.remainder;
> > +                Expr max_plus_one = max + 1;
> > +                max_plus_one -= bound.remainder;
> > +                max_plus_one = ((max_plus_one + bound.modulus - 1) / bound.modulus) * bound.modulus;
> > +                max_plus_one += bound.remainder;
> > +                extent = simplify(max_plus_one - min);
>
> Would it work to say the extent is just ((max - min + 1) / bound.modulus)
> * bound.modulus, where the min has already been aligned? It seems like this
> would produce the minimal extent required to cover the max, and would
> generate simpler code?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/halide/Halide/pull/1392/files/2cba09eb217797c8f39846ba246d0406c15ab993#r71436589>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAfdRlwx9ppVMn69-W7HrrACqTEpBlcOks5qXVB4gaJpZM4JQMmv>
> .
>